src/components/request/RequestForm.tsx

```diff
@@ -194,20 +215,76 @@ const RequestForm: FC = () => {
     }
   }, []);
 
+  // Temp. This throws if you try to read certain values too quickly.
+  // TODO: Add proper loader.
+  useEffect(() => {
+    setTimeout(() => {
+      setRaceCond(true);
+    }, 2000);
+  }, []);
+
   return (
     <RequestFormWrapper>
       <RequestFormRow>
         <RequestFormHeaderAndFormWrapper>
-          <FormHeader>Proposal</FormHeader>
+          <FormHeader>
+            {flags.InProposeState && "Proposal"}
+            {(flags.InDisputeState || flags.DisputeInProgress) &&
+              "Dispute Period"}
+          </FormHeader>
           <RequestFormInputWrapper>
             <RequestInputButtonBlock>
-              <RequestFormInput
-                label="Propose: "
-                value={value}
-                onChange={inputOnChange}
-              />
+              {flags.InProposeState && (
+                <RequestFormInput
+                  label="Propose: "
+                  value={value}
+                  onChange={inputOnChange}
+                />
+              )}
+              {flags.InDisputeState && raceCond && (
+                <RequestFormInput
+                  disabled={true}
+                  label="Proposed answer: "
+                  value={ethers.utils.formatUnits(
+                    read().request().proposedPrice,
+                    read().collateralProps().decimals
```

Comment:
I highly suggest you add this in `useOracleReader`, where collateral props and the current request already exist:
https://github.com/UMAprotocol/optimistic-oracle-dapp/blob/master/src/hooks/useOracleReader.ts#L20

This hook is also already in use in this file.  I would probably do the formatting in the hook and just export the variable `proposedPrice` directly already formatted to a float, to save you some logic in the view.